skcosmo/selection/CUR.py

```diff
@@ -0,0 +1,297 @@
+from abc import abstractmethod
+import numpy as np
+from scipy.sparse.linalg import eigs as speig
+
+from skcosmo.pcovr.pcovr_distances import get_Ct, get_Kt
+from .orthogonalizers import feature_orthogonalizer, sample_orthogonalizer
+
+
+class _CUR:
+    """
+    Base class for CUR selection methods
+    Requires a product, typically the gram or covariance matrix, \
+    from which to compute the importance score
+
+    If the model is iterative, the orthogonalize method must be overwritten
+    to orthogonalize the input matrices after each iteration.
+
+
+    :param iterative: whether to orthogonalize after each selection,
+                      defaults to `true`
+    :type iterative: boolean
+
+    :param k: number of eigenvectors to compute the importance score with,
+              defaults to 1
+    :type k: int
+
+    :param tolerance: threshold below which values will be considered 0,
+                      defaults to 1E-12
+    :type tolerance: float0
+    """
+
+    def __init__(self, iterative=True, tolerance=1e-12, k=1, **kwargs):
```

Comment:
There is no need to accept kwargs if we are doing nothing with them

```suggestion
    def __init__(self, iterative=True, tolerance=1e-12, k=1):
```